olive/model/hf_utils.py

```diff
@@ -25,6 +27,137 @@ class HFComponent(ConfigBase):
+        if not v:
+            return v
+
+        try:
+            return cls.dict_to_quantization_config(values["quantization_method"], v).to_dict()
+        except ImportError:
+            # we don't want to fail since the pass target might have the correct transformers version
+            logger.warning(
+                f"Could not import the config class for quantization method {values['quantization_method']}. Skipping "
+                " validation"
+            )
+            return v
+
+    def get_loading_args(self):
+        loading_args = {}
+        # copy args that can be directly copied
+        direct_copy_args = ["device_map", "max_memory"]
+        for arg in direct_copy_args:
+            if getattr(self, arg):
+                loading_args[arg] = deepcopy(getattr(self, arg))
+        # convert torch dtype to torch.dtype or "auto"
+        if self.torch_dtype:
+            loading_args["torch_dtype"] = self.get_torch_dtype()
+        # convert quantization_config to the config class
+        quantization_config = self.get_quantization_config()
+        if quantization_config:
+            loading_args["quantization_config"] = quantization_config
+        # add extra args
+        if self.extra_args:
+            loading_args.update(deepcopy(self.extra_args))
+        return loading_args
+
+    def get_torch_dtype(self):
+        v = self.torch_dtype
+        if isinstance(v, str) and v != "auto":
+            # get rid of torch. prefix, this might have been added when serializing
+            v = v.replace("torch.", "")
+            try:
+                return getattr(torch, v)
+            except AttributeError as e:
+                raise ValueError(f"Invalid torch dtype {v}") from e
+        return v
+
+    def get_quantization_config(self):
+        if not self.quantization_method or not self.quantization_config:
+            return None
+        return self.dict_to_quantization_config(self.quantization_method, self.quantization_config)
+
+    @staticmethod
+    def device_to_str(device):
+        if isinstance(device, torch.device):
+            device = str(device)
+        return device
+
+    @staticmethod
+    def dict_to_quantization_config(quantization_method, config_dict):
+        method_to_class_name = {"bitsandbytes": "BitsAndBytesConfig", "gptq": "GPTQConfig"}
+        method_to_min_version = {
+            # bitsandbytes exists from 4.27.0, but 4bit quantization is only supported from 4.30.0
+            "bitsandbytes": "4.30.0",
+            "gptq": "4.32.0",
+        }
+        if quantization_method not in method_to_class_name:
+            raise ValueError(
+                f"Unsupported quantization method {quantization_method}. Supported methods are"
+                f" {list(method_to_class_name.keys())}"
+            )
+
+        try:
+            config_cls = getattr(transformers, method_to_class_name[quantization_method])
```

Comment:
>getattr [](http://example.com/codeflow?start=25&length=7)

getattr will not raise exception. what's the except scenario here?

Reply:
`BitsAndBytesConfig` and `GPTQConfig` are not present in old versions. So I am using getattr and try/except to check for import error

Reply:
I suppose the old version will return None instead of raising exception for this case?

Reply:
![image](https://github.com/microsoft/Olive/assets/94929125/3c03e7d2-85a3-4eb5-9e11-b95d7732900e)

`getattr` raises AttributeError if a default value is not given.

Reply:
so, it seems the except is useless?